### PR TITLE
KFSPTS-28699 upgrade to the 10/19/2022 version of financials

### PR DESCRIPTION
**Pull request by db483**

### CODE REVIEW

**Comment by db483** [comment]
retest this please

File: pom.xml
Comment (cah292) [inline pom.xml:1]: It looks like KualiCo completely removed the commons-digester3 XML library for this upgrade, as per FINP-8140. However, We still appear to have some areas in our code that rely upon it, typically indicated by "...DigesterRules.xml" file definitions. If it's not practical to migrate them to a different technology at this time (such as JAXB), then I would suggest adding the commons-digester3 dependency to our POM and creating follow-up user stories to convert those areas over to JAXB.

I noticed at least the following areas in our code that still use the Digester3 library:

* Load Disbursement Vouchers Job (this is an rSmart mod and we might not be using it at the moment)
* I Want Document Feed Job
* Loading of PDP Payments into KFS (such as PDP Load Payments Job, possibly the related upload screen as well)
* Tax Processing Job

**Review by cah292 — CHANGES_REQUESTED** [review]
This is still under review, but in the meantime I noticed something important that will likely require further code or POM changes.

File: pom.xml
Comment (db483) [reply pom.xml:1]: Thank you, will update.

Comment (cah292) [reply pom.xml:1]: Regarding our customizations to the PDP payment loading, I'm not sure if our Digester3-related customization to that is even in use anymore, even prior to the 2022-10-19 changes. It seems that one of the earlier KualiCo releases may have already converted that over to JAXB. We may be able to remove our customization for that, unless there are sections of it that need to be carried over into the equivalent JAXB processing.

We'll still need to handle the other potential Digester3-dependent areas noted earlier.

File: src/main/resources/edu/cornell/kfs/pdp/cu-spring-pdp.xml
Comment (cah292) [inline src/main/resources/edu/cornell/kfs/pdp/cu-spring-pdp.xml:1]: While reviewing this file, I noticed that the `achPayeeLookupableHelperService` bean override is missing the "prototype" scope. Should that be fixed now or in a separate user story?

**Review by cah292 — COMMENTED** [review]
I noticed a couple other minor items that may or may not need changes at this time, but other than that (and apart from the item in my previous review) the code changes look good.

File: pom.xml
Comment (db483) [reply pom.xml:1]: Will investigate

File: src/main/java/edu/cornell/kfs/sys/batch/service/impl/CuBatchInputFileServiceImpl.java
Comment (db483) [reply src/main/java/edu/cornell/kfs/sys/batch/service/impl/CuBatchInputFileServiceImpl.java:69]: added

File: src/main/resources/edu/cornell/kfs/pdp/cu-spring-pdp.xml
Comment (db483) [reply src/main/resources/edu/cornell/kfs/pdp/cu-spring-pdp.xml:1]: Fixed although it is not related to changes in this upgrade.

File: pom.xml
Comment (db483) [reply pom.xml:1]: PaymentInputFileType still has a customization in the parse method for setting PayeeIdAndName and PayeeOwnerCdFromVendor so I think this still needs to be kept in place. Please let me know if you find otherwise. Thank you.

Comment (db483) [reply pom.xml:1]: I created KFSPTS-28860 to migrate the remaining batch jobs to jxb and added a customization to continue support for commons-digester3 in cu-kfs.

File: src/main/java/edu/cornell/kfs/sys/batch/service/DigestorXMLBatchInputFileType.java
Comment (cah292) [inline src/main/java/edu/cornell/kfs/sys/batch/service/DigestorXMLBatchInputFileType.java:21]: Would it be better to name this class `DigesterXmlBatchInputFileType` instead? (Note the changes in spelling and uppercasing.) I know some of the fields and arguments were using the "digestor" naming in base code, but I would recommend using the "digester" name for the class itself, for consistency with the Digester3 library that it uses.

Also, KualiCo updated `XmlBatchInputFileTypeBase` to use generics for specifying the parsed object type. I would suggest updating this class to either add generics usage or to have it extend `XmlBatchInputFileTypeBase<Object>`.

**Review by cah292 — CHANGES_REQUESTED** [review]
I noticed a few other items that may need to be addressed, but other than that the code changes look good.

File: src/main/java/org/kuali/kfs/pdp/batch/PaymentInputFileType.java
Comment (db483) [reply src/main/java/org/kuali/kfs/pdp/batch/PaymentInputFileType.java:1]: I think this can be removed now. Thank you.

File: pom.xml
Comment (db483) [reply pom.xml:417]: updated

File: src/main/java/edu/cornell/kfs/sys/batch/service/DigestorXMLBatchInputFileType.java
Comment (db483) [reply src/main/java/edu/cornell/kfs/sys/batch/service/DigestorXMLBatchInputFileType.java:21]: updated

**Comment by db483** [comment]
This is ready for review again.

**Review by cah292 — COMMENTED** [review]
There's one more minor follow-up item that may or may not need changes, but otherwise things look good.

File: src/main/java/edu/cornell/kfs/sys/batch/service/DigesterXMLBatchInputFileType.java
Comment (db483) [reply src/main/java/edu/cornell/kfs/sys/batch/service/DigesterXMLBatchInputFileType.java:20]: I generated the class name by adding Digester in front of the base class name. I see how it would make sense to change it to "Xml". Will update.

Will update the extend from.

The `outputClass` is not used by our Digester implementation but I will look to see if it makes sense to add it to the configuration for completeness. Thank you.

**Review by cah292 — COMMENTED** [review]
Latest changes look good, I just want to coordinate with the rest of the team before merging, given all the other pending PRs. We can discuss the timing at today's team standup.

**Comment by cah292** [comment]
Due to the testing efforts that will be taking place in the DEV environment this afternoon, I will wait until that testing is complete before merging the 2022-10-19 patch changes. If necessary, we could wait until tomorrow to merge the patch.

**Review by cah292 — APPROVED** [review]
Changes look good, unit tests pass and the SQL PR is also good to go. One minor revision is needed to the reports PR, but we can proceed with moving the patch to DEV in the meantime.

Merging.

**Pull request merged**
